patches/net/minecraft/world/entity/Entity.java.patch

```diff
@@ -326,6 +338,26 @@
 +        else return false;
 +    }
 +
++    public boolean updateFluidHeightAndDoCanPushEntityFluidPushing(boolean performFluidPushing) {
++        this.forgeFluidTypeHeight.clear();
```

Comment:
This should be fine, though I can't ever think of a situation where we don't check for fluids and clear the map anyway.

Reply:
Moved it into the performFluidPushing if statement